UniAtHome/UniAtHome.BLL/Services/CourseService.cs

```diff
@@ -7,38 +7,46 @@
 using UniAtHome.BLL.Interfaces;
 using UniAtHome.DAL.Entities;
 using UniAtHome.DAL.Interfaces;
+using UniAtHome.DAL.Repositories;
 
 namespace UniAtHome.BLL.Services
 {
     public class CourseService: ICourseService
     {
-        private readonly IRepository<Course> repository;
-
+        private readonly IRepository<Course> courseRepository;
+        private readonly IRepository<CourseMember> courseMemberRepository;
+        private readonly UserRepository userRepository;
         private readonly IMapper mapper;
 
-        public CourseService(IRepository<Course> repository, IMapper mapper)
+        public CourseService(IRepository<Course> courseRepository, IMapper mapper, IRepository<CourseMember> courseMemberRepository, UserRepository userRepository)
         {
-            this.repository = repository;
+            this.courseRepository = courseRepository;
             this.mapper = mapper;
+            this.courseMemberRepository = courseMemberRepository;
+            this.userRepository = userRepository;
         }
 
         public async Task<CourseDTO> GetCourseByIdAsync(int id)
         {
-            var course = await this.repository.GetByIdAsync(id);
-            return this.mapper.Map<CourseDTO>(course);
+            var course = await courseRepository.GetByIdAsync(id);
+            return mapper.Map<CourseDTO>(course);
         }
 
         public async Task AddCourseAsync(CourseDTO courseDto)
         {
-            var course = this.mapper.Map<Course>(courseDto);
-            await this.repository.AddAsync(course);
+            var course = mapper.Map<Course>(courseDto);
+            var newCourse = await courseRepository.AddAsync(course);
+            await courseRepository.SaveChangesAsync();
+            var teacherId = (await userRepository.FindByEmailAsync(courseDto.TeacherEmail)).Id;
+            await courseMemberRepository.AddAsync(new CourseMember(teacherId, newCourse.Id));
```

Comment:
I thought that we configured everything the way that there is no need to use courseMemberRepository and we can just use navigation property of our course entity (course.CourseMembers.Add(...))

Reply:
Fixed